src/Microsoft.Data/DataFrame.cs

```diff
@@ -131,7 +132,9 @@ public DataFrame Sort(string columnName, bool ascending = true)
             List<BaseColumn> newColumns = new List<BaseColumn>(ColumnCount);
             for (int i = 0; i < ColumnCount; i++)
             {
-                var newColumn = Column(i).Clone(sortIndices, !ascending);
+                BaseColumn oldColumn = Column(i);
+                BaseColumn newColumn = oldColumn.CloneAndAppendNulls(sortIndices, !ascending);
```

Comment:
`CloneAndAppendNulls` is internal. How could someone derive their own column type from BaseColumn, and then use this Sort method?

Reply:
Good catch!